[docs] Add a section on how to read from environment variables (#25782)
expo/expo: docs/pages/guides/environment-variables.mdx

Expo CLI loads **.env** files according to the [standard .env file resolution](https://github.com/bkeepers/dotenv/blob/c6e583a/README.md#what-other-env-files-can-i-use) and then replaces all references in your code to `process.env.EXPO_PUBLIC_[VARNAME]` with the corresponding value set in the **.env** files. Code inside **node_modules** is not affected for security purposes.

### How to read from environment variables

> - <YesIcon small /> **Every environment variable must be referenced as a static property for it to be

Reviewer: If possible, can a link be included explaining what a `static property` is referring to. I feel like this term isn't generally understood. If there is a good link that explains it that would be great to include here.

Author: I didn't find a direct link that explains the context of the environment variable as a static property but did add some context (see line 51) to explain what it means to be a static property in this context.

Reviewer: I'm good with this, someone can resolve this conversation as I cannot resolve it.

Reviewer 2: There are two different definitions of "static properties" here. The original documentation (not line 51) is talking about the property name being static. The extra clarification on line 51 is talking about the property value being static.

Furthermore, these values aren't runtime environment variables and they aren't loaded into the app's process. They are the build computer's environment variables and are inlined into the code. Will add a suggestion to this PR.

Expo CLI loads **.env** files according to the [standard .env file resolution](https://github.com/bkeepers/dotenv/blob/c6e583a/README.md#what-other-env-files-can-i-use) and then replaces all references in your code to `process.env.EXPO_PUBLIC_[VARNAME]` with the corresponding value set in the **.env** files. Code inside **node_modules** is not affected for security purposes.

### How to read from environment variables

> Once an environment variable is loaded into your app, it remains constant during that app's runtime. This means that the variable is a static property. It does not change unless a different value is set for the variable and the app is reloaded.
> 
> - <YesIcon small /> **Every environment variable must be referenced as a static property for it to be
>   inlined.** For example, the expression `process.env.EXPO_PUBLIC_KEY` is valid and will be inlined.

Reviewer 2: ```suggestion
- <YesIcon small /> **Every environment variable must be statically referenced as a property of `process.env` using JavaScript's dot notation for it to be inlined.** For example, the expression `process.env.EXPO_PUBLIC_KEY` is valid and will be inlined.
```

Author: Thanks for updating this and sharing the link, TIL!
